Prevent accidental inclusion of mpi c++ headers 
when hdf5.h is included third party library

https://bugs.gentoo.org/show_bug.cgi?id=420777
https://bugs.archlinux.org/task/33343

--- a/src/H5public.h
+++ b/src/H5public.h
@@ -57,6 +57,15 @@
 #   include <stddef.h>
 #endif
 #ifdef H5_HAVE_PARALLEL
+#   ifndef OMPI_SKIP_MPICXX
+#      define OMPI_SKIP_MPICXX
+#   endif
+#   ifndef MPICH_IGNORE_CXX_SEEK
+#      define MPICH_IGNORE_CXX_SEEK
+#   endif
+#   ifndef MPICH_SKIP_MPICXX
+#      define MPICH_SKIP_MPICXX
+#   endif
 #   include <mpi.h>
 #ifndef MPI_FILE_NULL		/*MPIO may be defined in mpi.h already       */
 #   include <mpio.h>
